### DISCUSSION TITLE
Kafka (RP or Confluent) available from other TC or host

### DISCUSSION THREAD

**Discussion by GregJohnStewart** [testcontainers/testcontainers-java#9122] [Q&A] [answered]
I have a situation where I need the kafka.* server from both another testcontainer and `localhost` (where the app I am testing exists). I am able to get connectivity working from `localhost`, but not the other testcontainer, as all the references that Red Panda advertises are all using `localhost`, and not the container's hostname. I have tried several ways to get this working, from `withListener` to setting environments to override the configured listener. In either case, the advertised listeners were still set to `localhost` and my other testcontainer isn't able to connect.

There should be a simple way to either consider `withNetworkAliases` or otherwise set the advertised hostnames. Also open to hearing what the group has to say on the matter. As stated, the `withListener` presented by the documentation doesn't seem to do the trick.

A complication of this is the host forwarded port isn't available at this configuration time.

Snippet of how I'm setting up my kafka instance:

```java
RedpandaContainer kafka = new RedpandaContainer(DockerImageName.parse("docker.redpanda.com/redpandadata/redpanda:v23.1.2"))
				.withNetwork(Network.SHARED)
				.withAccessToHost(true)
				.withNetworkAliases(KAFKA_DEVSERVICE_HOSTNAME)
//				.withListener(() -> "external://"+KAFKA_DEVSERVICE_HOSTNAME + ":9092")
				.withListener(() -> KAFKA_DEVSERVICE_HOSTNAME + ":9092")
				.withListener(() -> KAFKA_DEVSERVICE_HOSTNAME + ":9093")

				// { {external:{host: localhost, port: 32884}}, {internal:{host: 127.0.0.1, port: 9093}}} //snippet from logs of redpanda
//				.withEnv("REDPANDA_KAFKA_ADVERTISED_LISTENERS", String.format(
//						"internal://%s:9093,external://%s:9092",
//						KAFKA_DEVSERVICE_HOSTNAME,
//						KAFKA_DEVSERVICE_HOSTNAME
//					))
				;
```

**Comment by eddumelendez** [comment, marked as answer]
Hi, Have you checked the [Testcontainers Redpanda docs](https://java.testcontainers.org/modules/redpanda/) `Register additional listener` and `Produce/Consume via new listener`?

**Reply by GregJohnStewart** [reply]
TL;DR yes but now things are working so idunno. I'll blame being tired

**Reply by eddumelendez** [reply]
glad to hear is working and yeah taking a break always help.
